listen.gemspec

```diff
@@ -33,6 +33,9 @@ Gem::Specification.new do |s|
   s.add_dependency 'rb-fsevent', '>= 0.9.3'
   s.add_dependency 'rb-inotify', '>= 0.9.7'
 
+  # Used to show warnings at runtime
+  s.add_dependency 'ruby_dep', '~> 1.1'
```

Comment:
The Ruby version checking ("warnings") is not for dev mode. 

It's for runtime. E.g. if there's a ton of community pressure to make Listen compatible with Ruby < 2.2, then the Ruby requirement in the gemspec will be relaxed - but the warning about the outdated Ruby will remain.

(This is mostly so Rails users and web designers are encouraged to migrate to Ruby 2.3 - or at least avoid the buggy/insecure Ruby versions).

So ...

1. The version checking with RubyDep is mostly so people upgrade away from Ruby vulnerabilities at least. (Like upgrading to at least Ruby 2.2.4). That's what `RubyDep::Warning` is for: warning end users about their Rubies.

2. Adding `ruby_version` to the gem file is not really necessary. I don't know if Bundler imports [this gemspec line](https://github.com/guard/listen/blob/a2feb9d/listen.gemspec#L28) into the `Gemfile` as a `ruby_version` value. Ideally it should, but I don't know if it does. 

Since the `Gemfile` is only for development, it doesn't make sense to make it another place to set and track the Ruby version. (More confusion for contributors if different Ruby version constraints are in multiple places).

So this commit alone is just so end users are warned about Ruby vulnerabilities and bugs. There is a "bonus" in that it helps make sure the gem is present during development.

The other function of `ruby_dep` is to extract a list of supported Rubies from the `.travis.yml` files. It's a bit tricky to use an external gem to generate a `*.gemspec` files - because without that gem already installed, every Bundler command will fail during development.

The development dependency on `ruby_dep` is sufficient at 1.0.0. The warnings feature is in 1.1.0. I set both to 1.1.0 for consistency.